resource-managers/mesos/src/main/scala/org/apache/spark/scheduler/cluster/mesos/MesosClusterScheduler.scala

```diff
@@ -369,7 +369,8 @@ private[spark] class MesosClusterScheduler(
   }
 
   private def getDriverFrameworkID(desc: MesosDriverDescription): String = {
-    s"${frameworkId}-${desc.submissionId}"
+    val retries = desc.retryState.map{d => s"-retry-${d.retries.toString}"}
+    s"${frameworkId}-${desc.submissionId}${retries.getOrElse("")}"
```

Comment:
nit: move the `getOrElse()` call out of the string for clarity?

    val suffix = desc.retryState.map { }.getOrElse("")